library/std/src/sys/random/linux.rs

```diff
@@ -0,0 +1,153 @@
+// disable randomness for consistency. Otherwise, we'll try a raw syscall.
+// (`getrandom` was added in glibc 2.25, musl 1.1.20, android API level 28)
+syscall! {
+    fn getrandom(
+        buffer: *mut libc::c_void,
+        length: libc::size_t,
+        flags: libc::c_uint
+    ) -> libc::ssize_t
+}
+
+static GETRANDOM_AVAILABLE: AtomicBool = AtomicBool::new(true);
+static GRND_INSECURE_AVAILABLE: AtomicBool = AtomicBool::new(true);
+static URANDOM_READY: AtomicBool = AtomicBool::new(false);
+static DEVICE: OnceLock<File> = OnceLock::new();
+
+pub fn fill_bytes(mut bytes: &mut [u8], best_effort: bool) {
+    if GETRANDOM_AVAILABLE.load(Relaxed) {
+        loop {
+            if bytes.is_empty() {
+                return;
+            }
+
+            let flags = if best_effort {
+                if GRND_INSECURE_AVAILABLE.load(Relaxed) {
+                    libc::GRND_INSECURE
+                } else {
+                    libc::GRND_NONBLOCK
+                }
+            } else {
+                0
+            };
+
+            let ret = unsafe { getrandom(bytes.as_mut_ptr().cast(), bytes.len(), flags) };
+            if ret != -1 {
+                bytes = &mut bytes[ret as usize..];
+            } else {
+                match errno() {
+                    libc::EINTR => continue,
+                    // `GRND_INSECURE` is not available, try
+                    // `GRND_NONBLOCK`.
+                    libc::EINVAL if flags == libc::GRND_INSECURE => {
+                        GRND_INSECURE_AVAILABLE.store(false, Relaxed);
+                        continue;
+                    }
+                    // The pool is not initialized yet, fall back to
+                    // /dev/urandom for now.
+                    libc::EAGAIN if flags == libc::GRND_NONBLOCK => break,
+                    // `getrandom` is unavailable or blocked by seccomp.
+                    // Don't try it again and fall back to /dev/urandom.
+                    libc::ENOSYS | libc::EPERM => {
+                        GETRANDOM_AVAILABLE.store(false, Relaxed);
+                        break;
+                    }
+                    _ => panic!("failed to generate random data"),
+                }
+            }
+        }
+    }
+
+    // When we want cryptographic strength, we need to wait for the CPRNG-pool
+    // to become initialized. Do this by polling `/dev/random` until it is ready.
+    if !best_effort && !URANDOM_READY.load(Relaxed) {
+        let random = File::open("/dev/random").expect("failed to open /dev/random");
+        let mut fd = libc::pollfd { fd: random.as_raw_fd(), events: libc::POLLIN, revents: 0 };
+
+        while unsafe { libc::poll(&mut fd, 1, 0) == -1 } {
+            match errno() {
+                libc::EINTR => continue,
+                _ => panic!("failed to wait for /dev/urandom to become available"),
+            }
+        }
+
+        assert_eq!(fd.revents, libc::POLLIN);
+        URANDOM_READY.store(true, Relaxed);
```

Comment:
`File::open("/dev/urandom")` could be done here and stored instead of the temporary bool.

Reply:
See my other answer, that's not possible.